### PR TITLE
implement and test GroupedHistoryStates.

### PR DESCRIPTION
**Pull request by naoki-akimoto**
GroupedHistoryStatesの実装の修正とテストを追加しました、レビューをお願いします。

### CODE REVIEW

File: thingif/src/test/java/com/kii/thingif/query/GroupedHistoryStatesTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import com.kii.thingif.states.AirConditionerState;
+import com.kii.thingif.states.HumidityState;
+
+import junit.framework.Assert;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
```
Comment (yongpingchen) [inline thingif/src/test/java/com/kii/thingif/query/GroupedHistoryStatesTest.java]: please remove `import org.robolectric.annotation.Config;`

File: thingif/src/test/java/com/kii/thingif/query/GroupedHistoryStatesTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@RunWith(RobolectricTestRunner.class)
+public class GroupedHistoryStatesTest {
+
+    @Test
+    public void equals_hashCodeTest() {
+        TimeRange range = new TimeRange(new Date(1), new Date(100));
+        AirConditionerState air = new AirConditionerState();
+        HumidityState hum = new HumidityState();
+        Date createdAd = new Date(1);
+        HistoryState airHS = new HistoryState<AirConditionerState>(air, createdAd);
```
Comment (yongpingchen) [inline thingif/src/test/java/com/kii/thingif/query/GroupedHistoryStatesTest.java]: Please do not use raw type directly. 
```java
        HistoryState<AirConditionerState> airHS = new HistoryState<>(air, createdAd);
        HistoryState<HumidityState> humHS = new HistoryState<>(hum, createdAd);
        List<HistoryState<AirConditionerState>> list1 = new ArrayList<>();
        list1.add(airHS);
        List<HistoryState<HumidityState>> list2 = new ArrayList<>();
        list2.add(humHS);

        GroupedHistoryStates<AirConditionerState> target = new GroupedHistoryStates<>(range, list1);
        GroupedHistoryStates<AirConditionerState> sameOne = new GroupedHistoryStates<>(range, list1);
        GroupedHistoryStates<HumidityState> differentOne = new GroupedHistoryStates<>(range, list2);
```

**Comment by naoki-akimoto** [comment]
@yongpingchen  fixed. Please confirm.

**Comment by yongpingchen** [comment]
LGTM. The failure of CircleCI caused by other reason.

**Pull request merged**